crates/avian2d/examples/collision_layers.rs

```diff
@@ -2,11 +2,15 @@
 
 use avian2d::{math::*, prelude::*};
 use bevy::{prelude::*, sprite::MaterialMesh2dBundle};
-use examples_common_2d::XpbdExamplePlugin;
+use examples_common_2d::ExampleCommonPlugin;
 
 fn main() {
     App::new()
-        .add_plugins((DefaultPlugins, XpbdExamplePlugin))
+        .add_plugins((
+            DefaultPlugins,
+            ExampleCommonPlugin,
+            PhysicsPlugins::default().with_length_unit(15.0),
```

Comment:
I think a potential new user will be a bit confused by this line.
This goes for the other instances of `with_length_unit` in the other examples as well.

Reply:
These examples have a 2D camera with a default projection, so the objects use pixels for their dimensions and are quite large in terms of physics units if the length unit was the default `1.0`. This leads to the issues with overly soft contacts and incorrectly tuned tolerances, which can be seen at the end of the "Solver Results" section (the two screenshots).

Note that bevy_rapier also has this, but in 2D it calls the method [`pixels_per_meter`](https://github.com/dimforge/bevy_rapier/blob/56f98706751cb5cf74cb82e2b73749282c8ed5ee/bevy_rapier2d/examples/boxes2.rs#L13), even though internally the property is still called `length_unit`. And it also has a `with_length_unit` method, so it just has two differently named methods that essentially configure the same thing.

A quick survey on what names different engines use for this property:

- Rapier: [`length_unit`](https://docs.rs/rapier2d/latest/rapier2d/dynamics/struct.IntegrationParameters.html#structfield.length_unit) (or in 2D, also the `pixels_per_meter` method)
- Box2D: [`b2_lengthUnitsPerMeter`](https://github.com/erincatto/box2c/blob/1d7d1cf14722a06f6a7b1c3b0850e70bcbec8558/src/core.h#L110)
- PhysX: [`PxTolerancesScale.length`](https://nvidia-omniverse.github.io/PhysX/physx/5.1.0/docs/API.html#using-different-units)

I agree though that the method usage should probably be commented in the examples, or the method name itself could be more self-explanatory. It's just hard to come up with an approachable name that conveys the purpose, but does *not* imply that it actually scales velocities or forces in any way.

For example, if there was a `physics_scale` property and it was set to `10.0` (let's imagine this is 2D), a user could reasonably assume that `Collider::rectangle(1.0, 1.0)` would get scaled to a 10x10 pixel collider. Or you could assume that you could use `LinearVelocity::X`, and have the body move at the speed of 10 pixels per second. But this is not what `length_unit` does, as it just scales the internal length-based tolerances and thresholds to match the expected scale. The nuance is difficult to convey through the name of the property.

For these examples though, another option would be to just scale the camera projection so that physics objects don't have to be so large in terms of physics units and you don't need a custom length unit. This would technically be more ideal for physics in other ways too, although I don't know what the game dev best practices are for 2D games when it comes to using pixels vs. other units for scenes.

(sorry for the long response lol)

Reply:
For now, I added a comment briefly explaining the length unit to each example that sets it. I also improved the length unit docs overall